gapis/api/vulkan/state_rebuilder.go

```diff
@@ -461,11 +461,13 @@ func (sb *stateBuilder) write(cmd api.Cmd) {
 
 func (sb *stateBuilder) createInstance(vk VkInstance, inst InstanceObjectʳ) {
 	enabledLayers := []Charᶜᵖ{}
-	for _, layer := range inst.EnabledLayers().All() {
+	for _, k := range inst.EnabledLayers().Keys() {
+		layer := inst.EnabledLayers().Get(k)
```

Comment:
Whoa -- we must preserve the original order here! Layer order matters.

Reply:
I think this change actually fixes it, as the EnabledLayers here is a map of index to layer names, and they are stored in order during capture:

https://github.com/google/agi/blob/master/gapis/api/vulkan/api/instance.api#L120-L134

So following the order of this map keys (as we do here with ranging on `Keys()`) should respect the original order.

Reply:
:+1: